src/crypto/algorithms/megolm.js

```diff
@@ -279,21 +281,17 @@ MegolmEncryption.prototype._prepareNewSession = async function() {
  * @return {array<object<userid, deviceInfo>>}
  */
 MegolmEncryption.prototype._splitUserDeviceMap = function(
-    session, devicemap, devicesByUser,
+    session, chainIndex, devicemap, devicesByUser,
 ) {
     const maxToDeviceMessagesPerRequest = 20;
 
     // use an array where the slices of a content map gets stored
-    const contentMaps = [];
-    let maxContentMapId = 0; // start inserting in the first contentMap
-    let currentToDeviceId = 0;
-
-    const key = this._olmDevice.getOutboundGroupSessionKey(session.sessionId);
+    const mapSlices = [];
+    let currentSliceId = 0; // start inserting in the first slice
+    let entriesInCurrentSlice = 0;
 
-    for (const userId in devicesByUser) {
-        if (!devicesByUser.hasOwnProperty(userId)) {
-            continue;
-        }
+    for (let j = 0, userIds = Object.keys(devicesByUser); j < userIds.length; j++) {
```

Comment:
`for (const userId of Object.keys(devicesByUser))`

Sorry, think I said `in` where I meant `of` before